src/main/java/org/cloudfoundry/promregator/scanner/ReactiveAppInstanceScanner.java

```diff
@@ -182,9 +183,11 @@ public void setNumberOfInstances(int numberOfInstances) {
 				// NB: This drops the current target!
 				return Mono.empty();
 			}
-			
-			v.setApplicationId(sas.getId());
-			
+
+			if(v.getApplicationId() == null) {
```

Comment:
I am unsure whether this is necessary at all. If the `target` had not have an id in first place, then it would have been wrong by the `TargetResolver` to pass it through at all.  Hence, my impression is that this coding here is a little paranoid. 
Do you see a reason why it is necessary?

Reply:
Nope. It isn't necessary anymore. I'll delete it. Since this was my first contribution to the project I wasn't sure how paranoid you wanted me to be.

Reply:
Appreciated. 😁  
Paranoia isn't the worst habit for a developer, but that doesn't mean that we cannot rule things out...